magma/backend/mlir/build_magma_graph.py

```diff
@@ -100,19 +110,27 @@ def _visit_driver(
             src_module = _get_inst_or_defn_or_die(safe_root(ref.array.name))
             ctx.graph.add_edge(src_module, module, info=info)
             return
-        cache_key = (ref.array, ref.index)
+        index = ref.index
+        if isinstance(index, slice):
+            assert index.step is None
+            assert index.start <= index.stop
+            index = index.start, index.stop
+            getter_cls, getter_args = MagmaArraySliceOp, index
+        else:
+            getter_cls, getter_args = MagmaArrayGetOp, (index,)
+        cache_key = (ref.array, index)
```

Comment:
Maybe move this into a helper function `get_cache_key`

Reply:
not sure if what you meant, but factored out the get/set functionality